### PR TITLE
draft: cleanup flow logging for plugin/library use - v1

### PR DESCRIPTION
**Pull request by jasonish**
Some cleanups around the low level flow logging output as I was testing it from library/plugin context:
- remove unused arguments
- some renames to better form a public api
- should we put examples right in suricata proper?
- function level documentation
- TODO: devguide documentation

The idea is to give all the low level loggers the same treatment.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15967

File: src/output-flow.c [hunk 1]
```diff
@@ -39,33 +39,44 @@ typedef struct OutputFlowLoggerThreadData_ {
...
+ * \brief Register a new low-level flow logger.
+ *
+ * \param name The name of this logger. Its only used for debugging,
+ *     so choose something unique.
+ * \param initdata Data that will be passed to the ThreadInit
+ *     callback.
+ *
+ * \retval 0 on success, -1 on failure.
+ */
+int OutputRegisterFlowLogger(const char *name, SCFlowLoggerFunc LogFunc, void *initdata,
```
Comment (jasonish) [inline src/output-flow.c:67]: This is public API, should be `SC` prefixed.

File: src/output-flow.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ int OutputRegisterFlowLogger(const char *name, FlowLogger LogFunc,
 void OutputFlowShutdown(void);
 
 TmEcode OutputFlowLog(ThreadVars *tv, void *thread_data, Flow *f);
-TmEcode OutputFlowLogThreadInit(ThreadVars *tv, void *initdata, void **data);
+TmEcode OutputFlowLogThreadInit(ThreadVars *tv, void **data);
```
Comment (catenacyber) [inline src/output-flow.h:41]: Nice, do you have some tools to fine these simplifications ?

File: src/output-flow.h [hunk 3]
```diff
@@ -26,7 +26,9 @@
 #ifndef __OUTPUT_FLOW_H__
 #define __OUTPUT_FLOW_H__
 
+#include "decode.h"
```
Comment (catenacyber) [inline src/output-flow.h:29]: You can use `cppclean` to find these cf https://redmine.openinfosecfoundation.org/issues/945

**Comment by catenacyber** [comment]
I love this work :-)

Is there a redmine ticket to get it moving ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15967

**Comment by catenacyber** [comment]
Jason, what is the status of this ?

**Comment by jasonish** [comment]
Updated here: https://github.com/OISF/suricata/pull/11666

**Pull request closed without merging**